app/src/main/java/io/homeassistant/companion/android/settings/notification/NotificationViewModel.kt

```diff
@@ -0,0 +1,44 @@
+package io.homeassistant.companion.android.settings.notification
+
+import android.app.Application
+import android.app.NotificationManager
+import android.content.Intent
+import android.os.Build
+import android.provider.Settings
+import androidx.annotation.RequiresApi
+import androidx.compose.runtime.mutableStateListOf
+import androidx.core.content.getSystemService
+import androidx.lifecycle.AndroidViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+@RequiresApi(Build.VERSION_CODES.O)
+class NotificationViewModel @Inject constructor(
+    application: Application
+) : AndroidViewModel(application) {
+
+    val app = application
+
+    private var notificationManager = application.applicationContext.getSystemService<NotificationManager>()!!
+
+    var channelList = mutableStateListOf(notificationManager.notificationChannels)

```

Comment:
Right now this is a SnapshotStateList of the list of notification channels, is there any specific reason for that? Removing the inner list makes more sense to me, and it will also allow you to remove all the `[0]`s in the composable.

Reply:
Without this compose will not recompose when I call `updateChannelList`, even after I update it to clear all values.

Reply:
Any specific scenario you're testing where it doesn't work? When deleting a channel, and then recreating it using the 'Undo' option, I see it disappear and re-appear as expected. Code changed here to `notificationManager.notificationChannels.toMutableStateList()`, in the `updateChannelList` function to use `addAll` instead of `add`, and removed the `[0]` from the composable.

Reply:
ah I was missing `.toMutableStateList()`